src/loan/protocol_tests.rs

```diff
@@ -136,45 +112,62 @@ mod tests {
         let borrower = borrower.interpret(SECP256K1, loan_response).unwrap();
         let loan_transaction = borrower
             .sign({
-                let wallet = borrower_wallet.clone();
-                |transaction| async move { Ok(wallet.sign_all_inputs(transaction)) }
+                let wallet = wallet.clone();
+                |tx| async move {
+                    let wallet = wallet.lock().await;
+                    Ok(wallet.sign_inputs(tx))
+                }
             })
             .await
             .unwrap();
 
-        let loan_transaction = lender
+        // we don't need to verify that the loan transaction is valid
+        // because this library should have very little control over
+        // that: if either the lender or the borrower decide to use
+        // inputs they can't provide valid signatures for then this
+        // transaction won't be mined
```

Comment:
Does this make sense? Is there anything that we should be asserting on this transaction?

Reply:
Sounds good to me!

You could fake the inputs if you wanted. The spending script for a wallet-owned output is just a regular P2WPKH.

With these unit tests, you never have to proof the entire history of a coin, you can always start from where you care.

Not sure if there is much value though, if we can prove spending, chances of everything worked are pretty damn high.

Reply:
I've ended up verifying that these inputs are correctly signed for the loan/swap transaction.